NorthstarDLL/namedpipeclient.cpp

```diff
@@ -0,0 +1,140 @@
+#pragma once
+#include "pch.h"
+#include "namedpipeclient.h"
+#include "squirrel.h" // Squirrel
+
+// Named Pipe Stuff
+#include <windows.h>
+#include <conio.h>
+#include <tchar.h>
+#include <string>
+#include <atlstr.h>
+
+using namespace std;
+
+#define GENERAL_PIPE_NAME TEXT("\\\\.\\pipe\\GameDataPipe")
+#define BUFF_SIZE 512
+
+HANDLE hPipe;
+bool isConnected = false;
+
+ConVar* Cvar_ns_enable_named_pipe;
+
+HANDLE GetNewPipeInstance()
+{
+	HANDLE generalPipe = CreateFile(GENERAL_PIPE_NAME, GENERIC_READ, 0, nullptr, OPEN_EXISTING, 0, nullptr);
+	bool isGeneralConnected = generalPipe != INVALID_HANDLE_VALUE;
+	HANDLE thisMatchPipe = INVALID_HANDLE_VALUE;
+	if (isGeneralConnected)
+	{
+		TCHAR chBuf[BUFF_SIZE]; // Server message should either be BUFF_SIZE or Length of general pipe ids + terminating zero
+		DWORD cbRead;
+		if (ReadFile(
+				generalPipe, // pipe handle
+				chBuf, // buffer to receive reply
+				BUFF_SIZE * sizeof(TCHAR), // size of buffer
+				&cbRead, // number of bytes read
+				NULL)) // not overlapped
+		{
+			do
+			{
+				spdlog::info("New Pipe connection");
+				thisMatchPipe = CreateFile(chBuf, GENERIC_WRITE, 0, nullptr, OPEN_EXISTING, 0, nullptr);
+			} while (thisMatchPipe == INVALID_HANDLE_VALUE);
+		}
+	}
+	return thisMatchPipe;
+}
+
+void SendMessageToPipe(string message)
```

Comment:
Pass parameter "message" by const reference instead, no modification is performed.